src/github.com/matrix-org/dendrite/roomserver/api/query.go

```diff
@@ -0,0 +1,96 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/matrix-org/gomatrixserverlib"
+	"net/http"
+)
+
+// StateKeyTuple is a pair of an event type and state_key.
+type StateKeyTuple struct {
+	EventType     string
+	EventStateKey string
```

Comment:
Eh. I think using ptr to string is even more confusing since it would seem to imply that sending nil is a useful thing to do.